oauth2client/client.py

```diff
@@ -999,21 +999,17 @@ def _detect_gce_environment():
     #       could lead to false negatives in the event that we are on GCE, but
     #       the metadata resolution was particularly slow. The latter case is
     #       "unlikely".
-    connection = six.moves.http_client.HTTPConnection(
-        _GCE_METADATA_HOST, timeout=GCE_METADATA_TIMEOUT)
-
+    http = transport.get_http_object(timeout=GCE_METADATA_TIMEOUT)
```

Comment:
I would greatly prefer that, as when we switch to urllib3 we will want to use a single PoolManager instead of individual http instances.